service/qos/service_level_controller.cc

```diff
@@ -317,12 +317,18 @@ void service_level_controller::stop_update_from_distributed_data() {
     _global_controller_db->dist_data_update_aborter.request_abort();
 }
 
-future<std::optional<service_level_options>> service_level_controller::find_effective_service_level(auth::role_set roles, include_effective_names include_names) {
+future<std::optional<service_level_options>> service_level_controller::find_effective_service_level(const sstring& role_name) {
+    if (true) { //TODO: fix it, properly determine if service level controller already switched to raft updates
```

Comment:
It looks like we need the same condition as in your other PR (refreshing not every 10s but on every raft update) - check the cluster feature and the topology coordinator upgrade status.

Reply:
It's already done https://github.com/scylladb/scylladb/pull/18758/commits/976d97bc710afddd1abec2c7f23fa5b325e9f19e
I think `!group0_client.in_recovery() && feature_service.raft_service_levels_change && storage_service.get_topology_upgrade_state() == service::topology::upgrade_state_type::done` condition is enough in this place as well